### ISSUE TITLE
Error with serializer on typealias for primitive

### ISSUE THREAD

**Issue by zt64** [Kotlin/kotlinx.serialization#2083]
**Describe the bug**
I'm trying to use the typealias feature to be able to use the String class directly without having to access it through another property.
An error I get with the code I'm using is `Expected JsonPrimitive at text, found {"runs":[{"text":"All"}]}`
If I instead annotate the property with the serializer in the class constructor, then the deserialization works. But having to add this annotation dozens of times would not be very ideal.
I'm guessing that the String serializer is taking priority over the custom serializer that I've made.

**To Reproduce**
```kt
internal typealias ApiText = @Serializable(ApiTextSerializer::class) String

private class ApiTextSerializer : KSerializer<String> {
    override val descriptor: SerialDescriptor = buildClassSerialDescriptor("ApiText")

    override fun deserialize(decoder: Decoder): String {
        val input = decoder as JsonDecoder

        return input.decodeJsonElement().jsonObject["runs"]!!.jsonArray.joinToString(separator = "") {
            it.jsonObject["text"]!!.jsonPrimitive.content
        }
    }

    override fun serialize(encoder: Encoder, value: String) {
        throw RuntimeException("Serialization is not needed")
    }
}
```

**Expected behavior**
The deserialization process should not throw an error.

**Environment**
 - Kotlin version: 1.7.20
 - Library version: 1.4.1
 - Kotlin platforms: Android
 - Gradle version: 7.5.1

**Comment by sandwwraith** [comment]
Can you please post a stacktrace and sample class/data, please?

**Comment by sandwwraith** [comment]
I've checked it on the latest Kotlin master, and it seems to be fixed. I think it has the same root cause as https://github.com/Kotlin/kotlinx.serialization/issues/1895 — therefore, the fix will be available in Kotlin 1.8.20
